src/python/pants/backend/typescript/tsconfig.py

```diff
@@ -77,8 +78,39 @@ async def _read_parent_config(
     )
 
 
+def clean_tsconfig_contents(content: str) -> str:
```

Comment:
I first came up with a much simpler cleanup function:

```
    # Remove single-line comments (//) outside of string literals
    content = re.sub(r'(?<!["\'])//.*', '', content)

    # Remove multi-line comments (/* */) outside of string literals
    content = re.sub(r'/\*.*?\*/', '', content, flags=re.DOTALL)

    # Remove trailing commas
    # in objects: {"key": "value",}
    content = re.sub(r',\s*}', '}', content)

    # in arrays: ["value",]
    content = re.sub(r',\s*]', ']', content)
```

but have noticed that this won't work on `//` that might be present in URLs which led to much more complicated regular expressions.